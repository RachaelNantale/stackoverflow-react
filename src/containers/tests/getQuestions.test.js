import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ViewallQtnsConnected, { ViewallQtns } from '../getQuestions';

describe('Handles ViewAllQtns container', () => {
  let wrapper;
  let viewQuestions;
  const data = {
    title: 'what is this?',
    description: 'so what is this mahn?',
    tags: 'This',
  };
  const initialState = {
    getQuestionsReducer: { qtns: [] },
  };
  beforeEach(() => {
    viewQuestions = jest.fn();
    const mockStore = configureMockStore([thunk]);
    const store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <ViewallQtnsConnected
          questions={[data]}
          viewQuestions={viewQuestions}
        />
      </Provider>,
    );
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('renders correctly', () => {
    wrapper = mount(
      <ViewallQtns
        questions={[]}
        viewAllQtns={jest.fn()}
        viewQuestions={viewQuestions}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
