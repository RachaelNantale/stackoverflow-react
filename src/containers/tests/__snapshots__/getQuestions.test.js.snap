// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handles ViewAllQtns container renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(ViewallQtns)
    questions={
      Array [
        Object {
          "description": "so what is this mahn?",
          "tags": "This",
          "title": "what is this?",
        },
      ]
    }
    viewQuestions={[MockFunction]}
  >
    <ViewallQtns
      questions={Array []}
      viewAllQtns={[Function]}
      viewQuestions={[MockFunction]}
    >
      <div
        className="container"
      >
        Loading Questions ... 
      </div>
    </ViewallQtns>
  </Connect(ViewallQtns)>
</Provider>
`;

exports[`Handles ViewAllQtns container renders correctly 2`] = `
<ViewallQtns
  questions={Array []}
  viewAllQtns={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  viewQuestions={[MockFunction]}
>
  <div
    className="container"
  >
    Loading Questions ... 
  </div>
</ViewallQtns>
`;
